torch/csrc/autograd/python_cpp_function.cpp

```diff
@@ -151,7 +151,9 @@ PyObject* THPCppFunction_metadata(THPCppFunction* self, void* _unused) {
   auto metadata =
       static_cast<PyAnomalyMetadata*>(self->cdata->metadata())->dict();
 
-  Py_INCREF(metadata);
+  if (metadata) {
+    Py_INCREF(metadata);
+  }
```

Comment:
```suggestion
  Py_XINCREF(metadata);
```
is the idiomatic way of doing this in CPython.